src/workerd/api/queue.h

```diff
@@ -79,7 +81,15 @@ struct IncomingQueueMessage {
   kj::String id;
   kj::Date timestamp;
   kj::Array<kj::byte> body;
-  JSG_STRUCT(id, timestamp, body);
+  kj::Maybe<kj::String> contentType;
+  JSG_STRUCT(id, timestamp, body, contentType);
+
+  struct ContentType {
+    static constexpr kj::StringPtr TEXT = "text/plain"_kj;
+    static constexpr kj::StringPtr OCTET_STREAM = "application/octet-stream"_kj;
+    static constexpr kj::StringPtr JSON = "application/json"_kj;
+    static constexpr kj::StringPtr V8 = "application/v8"_kj;
```

Comment:
I fear this may be too simplistic. For one, content-types are case insensitive. `Application/Json` is as equally valid as `application/json`. It also doesn't take parameters into consideration, e.g. `application/json; charset=utf-8`. Given that `application/json` and `text/plain` are often used with the charset parameter it could end up being a source of bugs here.

Reply:
Fair points, I think ignoring case sensitivity would be easy enough. 

Handling the parameters seems a bit tougher -- this feature is taking inspiration from HTTP Content Types, but I don't know that we want/need to support all of the features -- the main goal is just to allow users to work in formats other than v8.

We control all the data going in/out of the Queue, so we could restrict to utf-8 perhaps. 

@a-robinson have any thoughts on this?

Reply:
We should avoid caring about case-sensitivity and just internally lower-case everything.

I don't really want to have to support parameters, although maybe I'm overestimating the complexity they'd add. I'd be happy to just shed the illusion that we support arbitrary HTTP content-types and instead use custom types (e.g. "text", "bytes", "json", "v8") if that'd be clearer than this halfway approach. I'm not sure we really benefit that much by attempting to reuse the standard content-type values.

What do you think would make for the best API, @jasnell ?